File: sample/src/main/java/de/blox/graphview/sample/MainContent.java
package de.blox.graphview.sample;

import java.util.ArrayList;
import java.util.List;

import de.blox.graphview.sample.Algorithms.BuchheimWalkerActivity;
import de.blox.graphview.sample.Algorithms.FruchtermanReingoldActivity;
import de.blox.graphview.sample.Algorithms.SugiyamaActivity;

public class MainContent {


    public static final List<GraphItem> ITEMS = new ArrayList<>();

    static {
        ITEMS.add(new GraphItem("BuchheimWalker", "Algorithm for drawing tree structures", BuchheimWalkerActivity.class));
        ITEMS.add(new GraphItem("FruchtermanReingold", "Directed graph drawing by simulating attraction/repulsion forces", FruchtermanReingoldActivity.class));
        ITEMS.add(new GraphItem("Sugiyama et al.", "Algorithm for drawing multilayer graphs, taking advantage of the hierarchical structure of the graph.", SugiyamaActivity.class));
    }

    public static class GraphItem {
        public final String title;
        public final String description;
        public final Class<?> clazz;

        public GraphItem(String title, String description, Class<?> clazz) {
            this.clazz = clazz;
            this.title = title;
            this.description = description;
        }

        @Override
        public String toString() {
            return title;
        }
    }
}

File: sample/src/main/java/dev/bandb/graphview/sample/MainContent.kt
package dev.bandb.graphview.sample

import dev.bandb.graphview.sample.algorithms.BuchheimWalkerActivity
import dev.bandb.graphview.sample.algorithms.FruchtermanReingoldActivity
import dev.bandb.graphview.sample.algorithms.SugiyamaActivity
import java.util.*

object MainContent {
    val ITEMS: MutableList<GraphItem> = ArrayList()

    class GraphItem(val title: String, val description: String, val clazz: Class<*>) {
        override fun toString(): String {
            return title
        }
    }

    init {
        ITEMS.add(GraphItem("BuchheimWalker", "Algorithm for drawing tree structures", BuchheimWalkerActivity::class.java))
        ITEMS.add(GraphItem("FruchtermanReingold", "Directed graph drawing by simulating attraction/repulsion forces", FruchtermanReingoldActivity::class.java))
        ITEMS.add(GraphItem("Sugiyama et al.", "Algorithm for drawing multilayer graphs, taking advantage of the hierarchical structure of the graph.", SugiyamaActivity::class.java))
    }
}